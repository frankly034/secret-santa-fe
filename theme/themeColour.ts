export enum PaletteColor {
  grey100 = "#1E1E1E",
  grey200 = "#282627",
  grey300 = "#3E4244",

  white = "#ffffff",
  white100 = "#e6e6e6",

  yellow100 = "#FDD365",
  red100 = "#F15156",
}

export enum ThemeDark {
  primaryTitleColor = PaletteColor.yellow100,
  secondaryTitleColor = PaletteColor.red100,

  appBackground = PaletteColor.grey100,
  controlBackground = PaletteColor.grey300,
  panelBackground = PaletteColor.grey200,

  foregroundColor = PaletteColor.white100,
}

export type Theme = "dark";

export type ThemeColor = PaletteColor | ThemeDark;
