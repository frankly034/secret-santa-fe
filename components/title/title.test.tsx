import { render } from "@testing-library/react";

import Title from ".";
import { PaletteColor } from "../../theme/themeColour";

describe("Component - Title", () => {
  it("renders title component", () => {
    const { container } = render(<Title>{"Gift a friend this holiday"}</Title>);
    expect(container).toMatchSnapshot();
  });

  it("renders single string child", () => {
    const { getByRole } = render(<Title>{"Gift"}</Title>);
    expect(getByRole("heading")).toBeTruthy;
  });

  it("should render last word with lastWordColor in a multiple string title", () => {
    const { getByText } = render(
      <Title secondaryColor={PaletteColor.white}>
        Gift a friend this holiday
      </Title>
    );
    expect(getByText("holiday").style.color).toEqual("rgb(255, 255, 255)");
  });

  it("should render single word title with preceeding color", () => {
    const { getByText } = render(
      <Title
        primaryColor={PaletteColor.yellow100}
        secondaryColor={PaletteColor.red100}
      >
        Gift
      </Title>
    );
    expect(getByText("Gift").style.color).toEqual("rgb(253, 211, 101)");
  });

  it("should render preceeding texts and last word with appropriate colors", () => {
    const { getByText } = render(
      <Title
        primaryColor={PaletteColor.white}
        secondaryColor={PaletteColor.red100}
      >
        Gift a friend this holiday
      </Title>
    );
    expect(getByText("Gift a friend this").style.color).toEqual(
      "rgb(255, 255, 255)"
    );
    expect(getByText("holiday").style.color).toEqual("rgb(241, 81, 86)");
  });
});
