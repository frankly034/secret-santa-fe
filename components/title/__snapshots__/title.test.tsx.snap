// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title - Component renders title component 1`] = `
<div>
  <h1
    class="ant-typography title css-dev-only-do-not-override-15rg2km"
  >
    <span
      style="color: rgb(253, 211, 101);"
    >
      Gift a friend this holiday
    </span>
  </h1>
</div>
`;
